src/renderer/containers/AddMetadataPage/index.tsx

```diff
@@ -72,66 +76,77 @@ export default function AddMetadataPage() {
     }, [dispatch, imagingSessions]);
 
     return (
-        <>
-            <div className={styles.contentContainer}>
-            {!isSelectedJobLoading && (
-            <>
-                {hasAttemptedSubmit && !appliedTemplate && (
-                <Alert
-                    className={styles.alert}
-                    message="Please select a template."
-                    type="error"
-                    showIcon={true}
-                    key="template-not-selected"
-                />
-                )}
-                <LabeledInput
-                className={styles.selector}
-                label={`Select Metadata ${SCHEMA_SYNONYM}`}
-                >
-                <TemplateSearch
-                    allowCreate={true}
-                    disabled={isTemplateLoading || isReadOnly}
-                    error={hasAttemptedSubmit && !appliedTemplate}
-                    value={appliedTemplate?.templateId}
-                    onSelect={(t) => dispatch(applyTemplate(t))}
-                />
-                </LabeledInput>
-            </>
-            )}
-            {isSelectedJobLoading ? (
-            <div className={styles.spinContainer}>
-                <div>Loading...</div>
-                <Spin />
+        <div>
+            <div>
+            {!selectedUploads.length && // If we're adding new files, not editing ones that have been uploaded.
+              <Button
```

Comment:
thoughts on testing a that the DOM matches state?  I thought React makes this sort of thing pretty convenient?

Reply:
You're right, I should add that. Would be extra nice if I could get [this ticket to change which library we use for DOM testing](https://github.com/aics-int/aics-file-upload-app/issues/144) done at some point too.